src/Castle.Core.Tests/Castle.Core.Tests-VS2017.csproj

```diff
@@ -1,22 +1,45 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
+	
 	<Import Project="..\..\buildscripts\common.props"></Import>
+
+	<PropertyGroup>
+		<TargetFrameworks Condition="'$(IsOSX)'=='true'">netcoreapp1.1</TargetFrameworks>
+		<TargetFrameworks Condition="'$(IsLinux)'=='true'">netcoreapp1.1</TargetFrameworks>
+		<TargetFrameworks Condition="'$(IsWindows)'=='true'">net461;netcoreapp1.1</TargetFrameworks>
+	</PropertyGroup>
+
+	<PropertyGroup>
+		<RuntimeIdentifier Condition="'$(IsOSX)'=='true'AND'$(TargetFramework)'=='net461'">osx.10.12-x64</RuntimeIdentifier>
+		<RuntimeIdentifier Condition="'$(IsOSX)'=='true'AND'$(TargetFramework)'=='netcoreapp1.1'"></RuntimeIdentifier>
+		<RuntimeIdentifier Condition="'$(IsLinux)'=='true'AND'$(TargetFramework)'=='net461'">ubuntu.16.04-x64</RuntimeIdentifier>
+		<RuntimeIdentifier Condition="'$(IsLinux)'=='true'AND'$(TargetFramework)'=='netcoreapp1.1'"></RuntimeIdentifier>
+		<RuntimeIdentifier Condition="'$(IsWindows)'=='true'">win7-x64</RuntimeIdentifier>
+	</PropertyGroup>
+	
 	<PropertyGroup>
-		<TargetFrameworks>net461;netcoreapp1.1</TargetFrameworks>
 		<OutputType>Exe</OutputType>
 		<GeneratePackageOnBuild>False</GeneratePackageOnBuild>
 		<AssemblyName>Castle.Core.Tests</AssemblyName>
 		<RootNamespace>CastleTests</RootNamespace>
-		<SignAssembly>True</SignAssembly>
+		 <SignAssembly Condition="'$(IsWindows)'=='true'">True</SignAssembly>
```

Comment:
No idea if it would help in this case by the `sn` tool used to be able to strong name an assembly, obviously only look at this if the `PublicSign` option doesn't work.